### ISSUE TITLE
Running legacy tasks from within v2 workflows in v1.4.2 with `slurm` backend seems broken

### ISSUE THREAD

**Issue by tcompa** [fractal-analytics-platform/fractal-server#1812]
This requires no action, as legacy-tasks-in-v2-workflows are deprecated as of v1.5.
But I'm very confused on whether we ever _successfully_ ran a legacy-task-in-v2-workflow job on a slurm instance.

---
```python traceback
UNKNOWN ERROR in Fractal job 455
TRACEBACK:
Traceback (most recent call last):
  File "/somewhere/fractal-server-env/lib/python3.10/site-packages/fractal_server/app/runner/v2/runner_functions.py", line 88, in _get_executor_options
    options = submit_setup_call(
  File "/somewhere/fractal-server-env/lib/python3.10/site-packages/fractal_server/app/runner/v2/_slurm_sudo/_submit_setup.py", line 75, in _slurm_submit_setup
    task_name=wftask.task.name,
AttributeError: 'NoneType' object has no attribute 'name'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/somewhere/fractal-server-env/lib/python3.10/site-packages/fractal_server/app/runner/v2/__init__.py", line 324, in submit_workflow
    new_dataset_attributes = await process_workflow(
  File "/somewhere/fractal-server-env/lib/python3.10/site-packages/fractal_server/app/runner/v2/_slurm_sudo/__init__.py", line 123, in process_workflow
    new_dataset_attributes = await async_wrap(_process_workflow)(
  File "/somewhere/fractal-server-env/lib/python3.10/site-packages/fractal_server/app/runner/async_wrap.py", line 25, in async_wrapper
    return await loop.run_in_executor(executor, pfunc)
  File "/somewhere/fractal-server-env/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/somewhere/fractal-server-env/lib/python3.10/site-packages/fractal_server/app/runner/v2/_slurm_sudo/__init__.py", line 79, in _process_workflow
    new_dataset_attributes = execute_tasks_v2(
  File "/somewhere/fractal-server-env/lib/python3.10/site-packages/fractal_server/app/runner/v2/runner.py", line 132, in execute_tasks_v2
    current_task_output = run_v1_task_parallel(
  File "/somewhere/fractal-server-env/lib/python3.10/site-packages/fractal_server/app/runner/v2/runner_functions.py", line 336, in run_v1_task_parallel
    executor_options = _get_executor_options(
  File "/somewhere/fractal-server-env/lib/python3.10/site-packages/fractal_server/app/runner/v2/runner_functions.py", line 96, in _get_executor_options
    raise RuntimeError(
RuntimeError: <class 'AttributeError'> error in submit_setup_call=<function _slurm_submit_setup at 0x7f9d615c9c60>
Original traceback:
  File "/somewhere/fractal-server-env/lib/python3.10/site-packages/fractal_server/app/runner/v2/runner_functions.py", line 88, in _get_executor_options
    options = submit_setup_call(
  File "/somewhere/fractal-server-env/lib/python3.10/site-packages/fractal_server/app/runner/v2/_slurm_sudo/_submit_setup.py", line 75, in _slurm_submit_setup
    task_name=wftask.task.name,
```

**Comment by jluethi** [comment]
> But I'm very confused on whether we ever successfully ran a legacy-task-in-v2-workflow job on a slurm instance.

I ran legacy task workflows successfully on a few workflows at UZH & FMI. A few FMI users also ran legacy tasks in May 2024 and they completed successfully

**Comment by jluethi** [comment]
I'm not sure anyone has run legacy tasks in V2 workflows since the 2.3.0 release (in mid July), so that could explain it

**Comment by tcompa** [comment]
Finally found it.

The bug was introduced within https://github.com/fractal-analytics-platform/fractal-server/pull/1523/files, see
![image](https://github.com/user-attachments/assets/b1947a32-546b-4169-b0e8-8edaafc6abbc)

This was the PR introducing per-task subfolders in the job main folder, and we used (a slug of) the task name for the subfolder name. This did not take into account legacy tasks, where the correct attribute would be `wftask.task_legacy.name` rather than `wftask.task.name`. It turns out that we were not testing legacy-tasks-in-v2-workflows for the SLURM backend.

That PR was released with v2.2.0, on June 20th.
